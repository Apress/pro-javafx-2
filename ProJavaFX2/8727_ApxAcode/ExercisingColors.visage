/*
 * @test
 * @run
 */

var a = Color.MAGENTA;
var b = Color.YELLOW;

println("Magenta = {a}");
println("Yellow = {b}");

println("Magenta + Yellow = {a + b}");
println("Magenta - Yellow = {a - b}");
println("Magenta * 2 = {a * 2}");
println("2 * Magenta = {2 * a}");
println("Magenta * 2.5 = {a * 2.5}");
println("2.5 * Magenta = {2.5 * a}");
println("Magenta / 2 = {a / 2}");
println("Magenta / 2.5 = {a / 2.5}");
println("Magenta / Yellow = {a / b}");

var navy = Color.valueOf(0x000080, false);
println("navy = {navy}");
var silver = Color.rgb(0xC0, 0xC0, 0xC0);
println("silver = {silver}");
var red = Color.color(1, 0, 0);
println("red = {red}");