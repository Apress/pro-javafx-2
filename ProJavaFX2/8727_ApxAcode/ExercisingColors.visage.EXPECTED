Magenta = color(100%, 0%, 100%)
Yellow = color(100%, 100%, 0%)
Magenta + Yellow = color(200%, 100%, 100%, 200%)
Magenta - Yellow = color(0%, -100%, 100%, 0%)
Magenta * 2 = color(200%, 0%, 200%, 200%)
2 * Magenta = color(200%, 0%, 200%, 200%)
Magenta * 2.5 = color(250%, 0%, 250%, 250%)
2.5 * Magenta = color(250%, 0%, 250%, 250%)
Magenta / 2 = color(50%, 0%, 50%, 50%)
Magenta / 2.5 = color(40%, 0%, 40%, 40%)
Magenta / Yellow = 1.0
navy = color(0%, 0%, 50%)
silver = color(75%, 75%, 75%)
red = color(100%, 0%, 0%)
